lib/rubocop/cop/metrics/utils/abc_size_calculator.rb

```diff
@@ -55,11 +54,26 @@ def evaluate_branch_nodes(node)
             end
           end
 
-          def node_has_else_branch?(node)
+          def evaluate_condition_node(node)
+            @condition += 1 if else_branch?(node)
+            @condition += 1
+          end
+
+          def else_branch?(node)
             %i[case if].include?(node.type) &&
```

Comment:
@jonas054 I suspect we could have a subtle bug here. We check for `#else_branch?` on `CONDITION_NODES`, which comes from `CyclomaticComplexity::COUNTED_NODES`, but that list does not include `case`:

```ruby
COUNTED_NODES = %i[if while until for
                   rescue when and or].freeze
```